src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs

```diff
@@ -255,14 +255,19 @@ public static IHostBuilder ConfigureDefaults(this IHostBuilder builder, string[]
                 });
 
             })
-            .UseDefaultServiceProvider((context, options) =>
+.UseDefaultServiceProvider((context, options) =>
             {
                 bool isDevelopment = context.HostingEnvironment.IsDevelopment();
                 options.ValidateScopes = isDevelopment;
                 options.ValidateOnBuild = isDevelopment;
             });
 
             return builder;
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+                Justification = "Calling IConfiguration.GetValue is safe when the T is bool.")]
+            static bool GetReloadConfigOnChangeValue(HostBuilderContext hostingContext) =>
+                            hostingContext.Configuration.GetValue("hostBuilder:reloadConfigOnChange", defaultValue: true);
```

Comment:
(nit) whitespacing here is a bit off. I would assume this line just has 1 level of indention.